schema/crdb/dbinit.sql

```diff
@@ -32,8 +32,8 @@ BEGIN;
  * NOTE: the database and user names MUST be kept in sync with the
```

Comment:
this is a lot of delta, which is fine of course, but if it's NOT all of the form `s/CREATE (.*) (.*)/CREATE \1 IF NOT EXISTS \2/` could you highlight OTHER changes?

Reply:
Almost all these changes are for idempotency and adding explicit names.

The only significant changes to dbinit.sql exist at the bottom, in regard to the `db_metadata` table, which has been re-worked.

Reply:
> Relatedly, there's something missing about tracking what state we're in. If we start applying an update and it fails for any reason (including, say, a complete rack power off), then the schema is wrong, in that it will report version X, when it's in fact at some version between version X and X + 1.

We actually are setting this up correctly. In the DB, we internally track a version, as well as a "target_version" field, which is only non-NULL when we're performing an update.

> I'm not sure if this is a problem right now or not, but certainly it would allow older Nexuses to start back up and think they're at the right version, when in fact they're not compatible with what's currently in the database. I can't see an easy way to address this without adding a second piece of metadata for "the version we're in the process of updating to".

See my comments here: https://github.com/oxidecomputer/omicron/pull/3760#discussion_r1275598857 - I spent a while ensuring that even if Nexus (or `schema-updater`, now) crashes and reboots, we should be able to safely re-apply these updates, and we don't try using structures from some other version until the whole transition completes successfully.